lib/Controller/DelegationController.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace OCA\GroupFolders\Controller;
+
+use OCP\IConfig;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Controller;
+use OCP\IGroup;
+use OCP\IGroupManager;
+use OCP\IRequest;
+use OCP\IUserManager;
+
+class DelegationController extends Controller {
```

Comment:
```suggestion
class DelegationController extends OCSController {
```

Let's make this an OCS Api right away :)

Reply:
I don't really understand NC'standpoint on the OCS API:

From the doc (https://docs.nextcloud.com/server/latest/developer_manual/basics/controllers.html#ocs), I read:

`This is purely for compatibility reasons. If you are planning to offer an external API, go for a api instead.`

Doesn't it means that we should avoid creating OCS API's? (What do you even mean by "api" in "go for a api instead"?)

If still, I should go to creating an OCS API, shouldn't I update this controller to return `DataResponse` instead of `JSONResponse`?

Thanks for your help

Reply:
Yes, should be a `DataResponse` then. The main reason for an OCS API here is that this can then be used both by the web ui with a logged in session as well as through basic auth. We might need to clarify that documentation a bit more.

Reply:
ok done